lib/runner.js

```diff
@@ -634,7 +640,12 @@ Runner.prototype.runTests = function(suite, fn) {
           self.emit(constants.EVENT_TEST_PENDING, test);
         }
         self.emit(constants.EVENT_TEST_END, test);
-        return next();
+        if (err instanceof Pending) {
+          self.suite = errSuite;
```

Comment:
This line is unnice, but the same technique is used already above in line549.
IMO, by just changing one property the runner gets inconsistent.

Reply:
I improved this part, it's the same pattern used by failing `beforeEach` hooks. 
Reviewed by myself.